daemon/cmd/daemon.go

```diff
@@ -930,6 +930,9 @@ func NewDaemon(ctx context.Context, cleaner *daemonCleanup, epMgr *endpointmanag
 		case option.Config.EgressMasqueradeInterfaces != "":
 			msg = fmt.Sprintf("BPF masquerade does not allow to specify devices via --%s (use --%s instead).",
 				option.EgressMasqueradeInterfaces, option.Devices)
+		case option.Config.EnableHostLegacyRouting:
+			msg = fmt.Sprintf("BPF masquerade requires full host reachable services enabled in tunnel mode but host legacy routing (--%s=\"true\")",
```

Comment:
Host reachable services (a.k.a. socket-level load balancing) and legacy host routing are two different things.

I think it would also be better to refactor this code so we don't have to keep the switch cases in sync. with the above if condition.

Reply:
@brb it appears you are familiar with this part of code and the background, what do you think about the refactor?

Reply:
it looks we can merge the if checks block and the switch case checks do we don't need to check the options twice.

Reply:
I made an initial refactor, please check if that is ok, I could not test it locally as it appears `make dev-docker-image` stop working anymore probably due to master branch changes.

Reply:
> I could not test it locally as it appears `make dev-docker-image` stop working anymore probably due to some master branch changes

It works fine for me (on commit 1059b958c8e9f9c88a810218ccc928900e4c98df). Are you on latest master? If you are and it persists, could you open a thread on Slack?